### PR TITLE
move config file to deploy stage

### PR DESCRIPTION
**Pull request by privacywill**
This PR solves https://github.com/manatee-project/manatee/issues/4
## Changes
- make the api and monitor to read configs from environment variables. 
- remove `pkg/config`.
- remove `pkg/conf` in bazel. 
- add a `--debug` parameters to `deploy.sh`. By default `debug` set to false. If the `debug` set to true, the confidential space won't be terminated, so the job status should always be `VM Running`.  You can delete the instance manually at gcp console.

## Caveats
- You should delete the old pod to make the new environment variable take effects. `kubectl delete pod -n $namespace` or `helm uninstall data-clean-room-helm -n $namespace`
## Testing
- Manual end-to-end testing. `./deploy.sh --namespace=xx --debug=true` works as expected.

### CODE REVIEW

**Comment by dayeol** [comment]
please make it a draft PR if it's WIP :)

**Comment by dayeol** [comment]
Please try to remove the global `Config` entirely, and have ReconcilerConfig (or TEEBackendConfig), and APIConfig

File: app/dcr_monitor/tee_backend/confidential_space.go [hunk 1]
```diff
@@ -27,7 +29,15 @@ type TEEProviderGCPConfidentialSpace struct {
 	client    *compute.InstancesClient
 }
 
-func NewTEEProviderGCPConfidentialSpace(ctx context.Context, projectId string, region string, zone string, env string) (*TEEProviderGCPConfidentialSpace, error) {
+func NewTEEProviderGCPConfidentialSpace(ctx context.Context) (*TEEProviderGCPConfidentialSpace, error) {
+	env := os.Getenv("ENV")
+	projectId := os.Getenv("PROJECT_ID")
+	region := os.Getenv("REGION")
+	zone := os.Getenv("ZONE")
```
Comment (dayeol) [inline app/dcr_monitor/tee_backend/confidential_space.go:36]: Can you please check if the environment variables are set, and print error if they are not present?

File: deployment/data-clean-room/values.yaml [hunk 2]
```diff
@@ -119,4 +119,11 @@ useMinikube: false
...
-namespace: ""
\ No newline at end of file
+namespace: ""
+
+config:
+  env: ""
+  project_id: ""
+  zone: ""
+  region: ""
+  debug: "false"
```
Comment (dayeol) [inline deployment/data-clean-room/values.yaml:129]: Please add newline

**Review by dayeol — CHANGES_REQUESTED** [review]
This is great, thanks!
Just left a few nits.

Also, could you run `bazel run //:gazelle`? Gazelle will automatically update BUILD files.

File: app/dcr_api/biz/service/job_service.go [hunk 3]
```diff
@@ -278,3 +277,8 @@ func (g *JobService) uploadFile(reader io.Reader, remotePath string, compress bo
 	}
 	return nil
 }
+
+func getBucket() string {
+	env := os.Getenv("ENV")
```
Comment (dayeol) [inline app/dcr_api/biz/service/job_service.go:282]: Please print warning/error when ENV is not set.

File: app/dcr_monitor/tee_backend/confidential_space.go (hunk 1 above)
Comment (privacywill) [reply app/dcr_monitor/tee_backend/confidential_space.go:45]: return err when the environment variables not set

File: app/dcr_api/biz/service/job_service.go (hunk 3 above)
Comment (privacywill) [reply app/dcr_api/biz/service/job_service.go:282]: only logs error when ENV is not set

**Review by dayeol — APPROVED** [review]
lgtm

**Pull request merged**